common/OpTestHMC.py

```diff
@@ -0,0 +1,348 @@
+            self.poweron_lpar(vios=True)
+
+    def poweroff_lpar(self):
+        if self.get_lpar_state() in [OpHmcState.NOT_ACTIVE, OpHmcState.NA]:
+            log.info('LPAR Already powered-off!')
+            return
+        self.run_command("chsysstate -m %s -r lpar -n %s -o shutdown --immed" %
+                         (self.system, self.lpar_name))
+        self.wait_lpar_state(OpHmcState.NOT_ACTIVE)
+
+    def poweron_lpar(self, runtime=False, vios=False):
+        if self.get_lpar_state(vios) == OpHmcState.RUNNING:
+            log.info('LPAR Already powered on!')
+            return
+        lpar_name = self.lpar_name
+        if vios:
+            lpar_name = self.lpar_vios
+        cmd = "chsysstate -m %s -r lpar -n %s -o on" % (self.system, lpar_name)
+
+        if not vios:
+            if self.lpar_prof:
+                cmd = "%s -f %s" % (cmd, self.lpar_prof)
+
+        self.wait_lpar_state(OpHmcState.NOT_ACTIVE, vios=vios)
+        self.run_command(cmd)
+        self.wait_lpar_state(vios=vios)
+        if runtime:
+            self.wait_login_prompt(self.get_console_prompt())
+            self.close_console(self.lpar_con)
+
+    def get_lpar_state(self, vios=False):
+        lpar_name = self.lpar_name
+        if vios:
+            lpar_name = self.lpar_vios
+        state = self.run_command(
+            'lssyscfg -m %s -r lpar --filter lpar_names=%s -F state' % (self.system, lpar_name))
+        return state[-1]
+
+    def get_system_state(self):
+        state = self.run_command(
+            'lssyscfg -m %s -r sys -F state' % self.system)
+        return state[-1]
+
+    def wait_lpar_state(self, exp_state=OpHmcState.RUNNING, vios=False, timeout=WAITTIME):
+        state = self.get_lpar_state(vios)
+        count = 0
+        while state != exp_state:
+            state = self.get_lpar_state(vios)
+            log.info("Current state: %s" % state)
+            time.sleep(timeout)
+            count = 1
+            if count > 120:
+                raise OpTestError("Time exceeded for reaching %s" % exp_state)
+
+    def wait_system_state(self, exp_state=OpManagedState.OPERATING, timeout=WAITTIME):
+        state = self.get_system_state()
+        count = 0
+        while state != exp_state:
+            state = self.get_system_state()
+            log.info("Current state: %s" % state)
+            time.sleep(timeout)
+            count = 1
+            if count > 60:
+                raise OpTestError("Time exceeded for reaching %s" % exp_state)
+
+    def console_run_command(self, console, cmd, timeout=60):
+        if console is None:
+            console = self.get_console()
+
+        try:
+            console.send(cmd)
+            console.send('\r')
+            output = console.before
+            console.sendline("echo $?")
+            console.send('\r')
+            # Time to accumulate both outputs
+            time.sleep(STALLTIME)
+            rc = console.expect(["\[console-pexpect\]#$"], timeout=timeout)
```

Comment:
Suggest using the utility to build the prompt to provide a system wide common abstraction.

Reference common/OpTestIPMI.py for a model.

OpTestHMC __init__ add the prompt=None

self.util = OpTestUtil()
self.prompt = prompt
self.expect_prompt = self.util.build_prompt(prompt) + "$"